solutions/USACO jan16-fortmoo.mdx

```diff
@@ -0,0 +1,122 @@
+---
+id: usaco-600
+source: USACO Gold 2016 January
+title: Fort Moo
+author: Kevin Sheng
+---
+
+[Official Editorial](http://www.usaco.org/current/data/sol_fortmoo_platinum_jan16.html)
+
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+#include <iostream>
+#include <fstream>
+#include <vector>
+#include <algorithm>
+
+using std::cout;
+using std::endl;
+using std::vector;
+
+const char BAD = 'X';
+
+// 2016 jan plat
+int main() {
+	std::ifstream read("fortmoo.in");
+	int row_num;
+	int col_num;
+	read >> row_num >> col_num;
+
+	vector<vector<bool>> sturdy(row_num, vector<bool>(col_num));
+	vector<vector<int>> row_bad_nums(row_num, vector<int>(col_num + 1));
+	for (int r = 0; r < row_num; r++) {
+		for (int c = 0; c < col_num; c++) {
+			char cell;
+			read >> cell;
+			sturdy[r][c] = cell != BAD;
+			row_bad_nums[r][c + 1] += row_bad_nums[r][c] + (cell == BAD);
+		}
+	}
+
+	int max_area = 0;
+	for (int c_start = 0; c_start < col_num; c_start++) {
+		for (int c_end = c_start + 1; c_end < col_num; c_end++) {
+			int r_start = 0;
+			for (int r_end = 0; r_end < row_num; r_end++) {
+				bool rowValid = row_bad_nums[r_end][c_end + 1] - row_bad_nums[r_end][c_start] == 0;
+				if (r_end == r_start && !rowValid) {
+					r_start++;
+					continue;
+				}
+				if (!sturdy[r_end][c_start] || !sturdy[r_end][c_end]) {
+					r_start = r_end + 1;
+				} else if (rowValid && r_start != r_end) {
+					max_area = std::max(max_area, (c_end - c_start + 1) * (r_end - r_start + 1));
+				}
+			}
+		}
+	}
+	cout << max_area << endl;
```

Comment:
i don't think this is necessary?

Reply:
This way, when people run the code, they won't have to constantly switch between files to check output.

Reply:
Maybe use `cerr` instead? `cout` is fine though